File: core/src/ca/kanoa/zombieworld/events/EventHandler.java
package ca.kanoa.zombieworld.events;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Target(ElementType.METHOD)
@Retention(RetentionPolicy.RUNTIME)
public @interface EventHandler {

    EventPriority eventPriority() default EventPriority.DEFAULT;

    boolean ignoreCancelled() default true;

}

File: core/src/ca/kanoa/zombieworld/events/EventHandler.kt
package ca.kanoa.zombieworld.events

import java.lang.annotation.Retention
import java.lang.annotation.RetentionPolicy

@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
@Retention(RetentionPolicy.RUNTIME)
annotation class EventHandler(val eventPriority: EventPriority = EventPriority.DEFAULT, val ignoreCancelled: Boolean = true)
